specs/_features/eip7251/beacon-chain.md

```diff
@@ -0,0 +1,887 @@
+    set_or_append_list(state.balances, index, 0)  # [Modified in EIP7251]
+    set_or_append_list(state.previous_epoch_participation, index, ParticipationFlags(0b0000_0000))
+    set_or_append_list(state.current_epoch_participation, index, ParticipationFlags(0b0000_0000))
+    set_or_append_list(state.inactivity_scores, index, uint64(0))
+    state.pending_balance_deposits.append(PendingBalanceDeposit(index, amount))  # [New in EIP7251]
+```
+
+###### Updated `get_validator_from_deposit`
+
+```python
+def get_validator_from_deposit(pubkey: BLSPubkey, withdrawal_credentials: Bytes32) -> Validator:
+    return Validator(
+        pubkey=pubkey,
+        withdrawal_credentials=withdrawal_credentials,
+        activation_eligibility_epoch=FAR_FUTURE_EPOCH,
+        activation_epoch=FAR_FUTURE_EPOCH,
+        exit_epoch=FAR_FUTURE_EPOCH,
+        withdrawable_epoch=FAR_FUTURE_EPOCH,
+        effective_balance=0,  # [Modified in EIP7251]
+    )
+```
+
+##### Withdrawals 
+
+###### New `process_execution_layer_withdraw_request`
+
+```python
+def process_execution_layer_withdraw_request(
+    state: BeaconState,
+    execution_layer_withdraw_request: ExecutionLayerWithdrawRequest
+) -> None:
+    amount = execution_layer_withdraw_request.amount
+    is_full_exit_request = amount == 0
+    # If partial withdrawal queue is full, only full exits are processed 
+    if not (is_full_exit_request or len(state.pending_consolidations) < PENDING_PARTIAL_WITHDRAWALS_LIMIT):
+        return
+
+    validator_pubkeys = [v.pubkey for v in state.validators]
+    index = ValidatorIndex(validator_pubkeys.index(execution_layer_withdraw_request.validator_pubkey))
+    validator = state.validators[index]
+
+    # Same conditions as in EIP-7002
+    if not (
+        has_execution_withdrawal_credential(validator)
+        # Verify withdrawal credentials
+        and validator.withdrawal_credentials[12:] == execution_layer_withdraw_request.source_address
+        # Verify the validator is active
+        and is_active_validator(validator, get_current_epoch(state))
+        # Verify exit has not been initiated, and slashed
+        and validator.exit_epoch == FAR_FUTURE_EPOCH
+        # Verify the validator has been active long enough
+        and get_current_epoch(state) >= validator.activation_epoch + SHARD_COMMITTEE_PERIOD
+    ):
+        return
+    # New condition: only allow partial withdrawals with compounding withdrawal credentials
+    if not (is_full_exit_request or has_compounding_withdrawal_credential(validator)):
+        return
+
+    pending_balance_to_withdraw = sum(
+        item.amount for item in state.pending_partial_withdrawals if item.index == index
+    )
+    # only exit validator if it has no pending withdrawals in the queue
+    if is_full_exit_request and pending_balance_to_withdraw == 0:
+        initiate_validator_exit(state, index)
+    elif state.balances[index] > MIN_ACTIVATION_BALANCE + pending_balance_to_withdraw:
```

Comment:
```suggestion
    elif state.balances[index] >= MIN_ACTIVATION_BALANCE + pending_balance_to_withdraw:
```

Reply:
In the equality case there is nothing left to withdraw so the withdrawal shouldn’t be created